src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -618,15 +618,12 @@ static int CountDigits(int value)
         }
 
         // Some fields that have a reserved spot in the header, may not fit in such field anymore, but they can fit in the
-        // extended attributes. They get collected and saved in that dictionary, with no restrictions.
-        private void CollectExtendedAttributesFromStandardFieldsIfNeeded()
+        // extended attributes. They are always collected or updated in that dictionary, with no restrictions.
+        private void CollectExtendedAttributesFromStandardFields()
         {
-            ExtendedAttributes.Add(PaxEaName, _name);
+            ExtendedAttributes[PaxEaName] = _name;
 
-            if (!ExtendedAttributes.ContainsKey(PaxEaMTime))
-            {
-                ExtendedAttributes.Add(PaxEaMTime, TarHelpers.GetTimestampStringFromDateTimeOffset(_mTime));
-            }
```

Comment:
This behavior looks like it was intentional, is it OK that we change it from "add if it was absent" to "always set"?

Reply:
This `CollectExtendedAttributesFromStandardFields` method does exactly what other tar tools do when creating a PAX entry:
- They always insert the values of `name` and `mtime` for all entry types, regardless if they fit or not in the standard field.
- They insert `linkname` for symlinks and hardlinks if not empty, regardless if it fits or not in the standard field.
- If `gname` or `uname` are set, they get added if their byte lengths are too large for the standard field.
- If `size` does not fit in the standard field when converted to string, it gets inserted.

If the user decides to manually change the `mtime` before writing it into the TarWriter, then the `mtime` would not be updated.

I need to add more unit tests.

Reply:
Addressed.